File: caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    if (argc != 2) //number of argument is two.
    {
        printf("number of argument is two.\n");
        return 1;
    }
    
    
    for (int i = 0, charlen = strlen(argv[1]); i < charlen; i++) //loop on length of argv[1].
    {
        if (!(isdigit(argv[1][i]))) //argv[1] is integer.
        {
            printf("argc is integer.\n");
            return 1;
        }
    }
    
    int k = argc; //number of shifting
    string plaintext = get_string("plaintext is: "); //prompt plaintext
    printf("ciphertext: ");
    
    for (int i = 0, textLen = strlen(plaintext); i < textLen; i++) //loop in the length of text
    {
        int asciiNum = plaintext[i]; //convert plaintext to ASCII code
        if (asciiNum >= 'A' && asciiNum <= 'Z') //case of uppercase letter
        {
            asciiNum = (asciiNum + k - 'A') % 26 + 'A'; //shift by k
        }
        else if (asciiNum >= 'a' && asciiNum <= 'z') //case of lowercase letter
        {
            asciiNum = (asciiNum + k - 'a') % 26 + 'a'; //shift by k
        }
        printf("%c", asciiNum); //convert shifted number array for encrypted message.
    }
    printf("\n");
}

File: caesar.py
import sys

args = sys.argv
argc = len(args)


def main():

    if (argc != 2):  # number of argument is two.
        print("number of argument is two.\n")
        sys.exit()

    for i in args[1]:  # loop on length of args[1]
        if not(str.isdigit(i)):  # args[1] is integer.
            print("argc is integer.\n")
            sys.exit()

    k = int(args[1])
    plaintext = None
    ciphertext = ''
    while (plaintext == None or plaintext == ''):
        plaintext = input('plaintext is...  ')

    for j in plaintext:   # loop in the length of text
        asciiNum = ord(j)   # convert plaintext to ASCII code
        if (asciiNum >= ord('A') and asciiNum <= ord('Z')):  # case of uppercase letter
            asciiNum = (asciiNum + k - ord('A')) % 26 + ord('A')  # shift by k
        elif (asciiNum >= ord('a') and asciiNum <= ord('z')):  # case of lowercase letter
            asciiNum = (asciiNum + k - ord('a')) % 26 + ord('a')  # shift by k
        ciphertext += chr(asciiNum)

    print('ciphertext: ' + ciphertext + '\n')  # convert shifted number array for encrypted message.

    return 0


main()
